spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/retry/RetryTemplateConfigurer.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2012-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.autoconfigure.retry;
+
+import org.springframework.retry.support.RetryTemplate;
+
+/**
+ * A configurer for a {@link RetryTemplate}, for example, to add a RetryListener.
+ *
+ * @param <P> the properties type.
+ *
+ * @author Gary Russell
+ * @since 2.0
+ *
+ */
+public interface RetryTemplateConfigurer<P> {
+
+	/**
+	 * Configure the template.
+	 * @param template the template.
+	 * @param properties the properties.
+	 */
+	void configure(RetryTemplate template, P properties);
```

Comment:
There is a mixed of concept here. Usually a customizer is something that  applies something extra. It doesn't  get the configuration properties object in argument (as it is an auto-configuration concern).